docs/06-using-cairo.md

```diff
@@ -0,0 +1,160 @@
+
+**Surface**:
+
+A surface is the area where the painting will be done. There is multiple types
+of surfaces including:
+
+ * Color pixmap with transparency (`ARGB32`) or without (`RGB24`)
+ * Monochrome pixmap surfaces with transparency (`A8`) or without (`A1`)
+ * SVG vectorial surfaces
+ * Native (XCB) surfaces
+ * Framebuffers and other less interesting ones from an AwesomeWM point of view.
+
+For more details, read:
+
+* [Surfaces](https://cairographics.org/manual/cairo-Image-Surfaces.html).
+
+**Sources**:
+
+Sources are elements like colors, patterns or gradient. See `gears.color` for
+common sources.
+
+For more details, read:
+
+* [Pattern](http://cairographics.org/manual/cairo-cairo-pattern-t.html)
+
+**Context and paths**:
+
+A context is the proxy between the program and the surface. The context hold a
+set of paths. Paths are something like a line, circle or rectangle. They may or
+may not be closed (a shape). All operations on the surfaces has to be told to
+the context. Until the context state is flushed (see next section), those paths
+add up, until that, nothing will be written to the surface. For example:
+
+    cr:rectangle(0 , 0 , 10, 10)
+    cr:rectangle(10, 10, 10, 10)
+
+wont do anything until and operation is applied on the context.
+
+For more details, read:
+
+* [Path](http://cairographics.org/manual/cairo-Paths.html)
+* [Context](http://cairographics.org/manual/cairo-cairo-t.html)
+* [Transformation](http://cairographics.org/manual/cairo-Transformations.html)
+
+A context also hold a transformation matrix (see `gears.matrix`) used when
+applying an operation.
+
+**Operations**:
+
+Multiple operations can be done with the paths. The most common are:
+
+ * *fill*: Fill the path with the current source
+ * *stroke*: Paint the path outline with the current source
+ * *mask*: Use the current source as an alpha mask while painting with the
+   current operator.
+ * *clip*: Crop the surfaces workarea so nothing outside of the clip will be
+   affected by the next opetations.
+
+**Operators**:
+
+[Operators](http://cairographics.org/operators/) are modifiers used when
+applying operations.
+
+### Cairo in Awesome
+
+Each Awsome `wibox`es, `awful.wibar`s, `gears.wallpaper`s and `awful.titlebar`s
+are in fact glorified Cairo surfaces. They can be used as such using an hidden
+API called `drawin`. Howver, they let widgets use the Cairo context directly.
+See the
+[declarative layout system](../documentation/03-declarative-layout.md.html)
+and [new widgets](../documentation/04-new-widgets.md.html) articles for more
+information and examples.
+
+It is also possible to create surfaces manually. See `gears.surface` for
+some examples. Here is the most simple example you can get:
+
+    -- Create a surface
+    local img = cairo.ImageSurface.create(cairo.Format.ARGB32, 50, 50)
+
+    -- Create a context
+    local cr  = cairo.Context(img)
+
+    -- Set a red source
+    cr:set_source(gears.color("#ff0000"))
```

Comment:
Since you mention "most simple": Can you make this `cr:set_source_rgb(1, 0, 0)`?